### DISCUSSION TITLE
Generic kubernetes client in the IKubernetesClient interface in v8

### DISCUSSION THREAD

**Discussion by slacki123** [dotnet/dotnet-operator-sdk#705] [Q&A] [answered]
Hi, I've started using the new v8 release recently and noticed that the KubeOps IKubernetesClient interface no longer has access to the generic Kubernetes. Is this intentional, so that we are supposed to use the Kubeops client, and the k8s client separately, or is this an omission? Otherwise what's the best practice use to use functions such as Patching or other generic k8s client methods?

**Comment by buehler** [comment]
Hey @slacki123 

What do you mean specifically? The `GenericClient<>` from the original client?
It is not intentionally omitted. I restructured everything for v8.

**Reply by slacki123** [reply]
Sorry I mean the previous `IKubernetesClient` had the `ApiClient` property which was the `IKubernetes` interface with all the useful kubernetes APIs. Now `IKubernetes` is not accessible from `IKubernetesClient`

So now if I want to use `Patch` functionality, I have to inject `IKubernetes` separately

**Reply by slacki123** [reply]
I guess what I'm trying to ask is whether we should be separately injecting and using IKubernetes, whether it should be accessible on the API client, or should we add more IKubernetesClient methods such as patch

**Reply by buehler** [reply]
Hmm good question.

If it's only patch you are missing, I think we can just add patch. However, if more stuff should be added, I'd suggest to use the IKubernetes client. It should not interfere with other clients.

Or would you rather use the IKubernetesClient?

**Reply by slacki123** [reply]
so right now I use the interfaces like this

```
 public MyClass(
        IKubernetesClient kubeOpsKubernetesClient,
        IKubernetes kubernetes,
       )
```

So this feels rather weird. And also because IKubernetesClient relies on IKubernetes anyway, it feels like the KubernetesClient methods should be just extensions for IKubernetes in my opinion

**Reply by buehler** [reply]
fair enough :-)

**Reply by slacki123** [reply]
also it looks like Patch is already a method available on the Generic Client. Any idea why it hasn't been added to kubeops IKubernetesClient already?

https://github.com/kubernetes-client/csharp/blob/master/src/KubernetesClient/GenericClient.cs#L87

**Reply by buehler** [reply, marked as answer]
I just used replace everytime, so this was a personal opinion
